tiled/client/composite.py

```diff
@@ -84,39 +84,73 @@ def create_composite(self, key=None, *, metadata=None, specs=None):
         """Composite nodes can not include nested composites by design."""
         raise NotImplementedError("Cannot create a composite within a composite node.")
 
-    def read(self, variables=None):
+    def read(self, variables=None, dim0=None):
+        """Download the contents of a composite node as an xarray.Dataset.
+
+        Args:
+            variables (list, optional): List of variable names to read. If None, all
+                variables are read. Defaults to None.
+
+        Returns:
+            xarray.Dataset: The dataset containing the requested variables.
+        """
+        import pandas
         import xarray
 
-        if variables is None:
-            # Read all.
-            data_vars = {}
-            for part, item in self.get_contents().items():
-                if item["attributes"]["structure_family"] != StructureFamily.table:
+        data_vars = {}
+        for part, item in self.get_contents().items():
+            # Read all or selective arrays/columns.
+            if item["attributes"]["structure_family"] in {
+                StructureFamily.array,
+                StructureFamily.sparse,
+            }:
+                if variables is None or part in variables:
```

Comment:
Purely for readability (no change in behavior)

```suggestion
                if (variables is None) or (part in variables):
```